src/main/java/pt/dinis/client/login/LoginClientScannerProtocol.java

```diff
@@ -0,0 +1,234 @@
+package pt.dinis.client.login;
+
+import org.apache.log4j.Logger;
+import pt.dinis.main.Display;
+
+import java.util.*;
+
+/**
+ * Created by tiago on 22-01-2017.
+ */
+public class LoginClientScannerProtocol {
+
+    public enum MessageType {
+        HELP("help", "Displays this message",
+                "help", Arrays.asList("h")),
+        INFO("info", "Shows client information",
+                "info", Arrays.asList("information", "status")),
+        LOGIN("login", "Logs in the server",
+                "login; login ip; login ip port", Collections.emptyList()),
+        LOGOUT("logout", "Logs out of the server",
+                "logout", Collections.emptyList()),
+        RELOGIN("relogin", "Redo the login",
+                "relogin", Arrays.asList("re-login")),
+        START("start", "Starts a communication with the server",
+                "start", Arrays.asList("open", "connect")),
+        CLOSE("close", "Closes a communication with the server",
+                "close", Arrays.asList("disconnect")),
+        MESSAGE("message", "Sends a message to the server",
+                "message text; text", Collections.emptyList()),
+        HASH("hash", "Print hash given from server while logging in",
+                "hash", Collections.emptyList()),
+        EXIT("exit", "Ends this client",
+                "exit", Arrays.asList("quit", "end"));
+
+        private String word;
+        private String help;
+        private String usage;
+        private Collection<String> alternatives;
+
+        MessageType(String word, String help, String usage, Collection<String> alternatives) {
+            this.word = word;
+            this.help = help;
+            this.usage = usage;
+            this.alternatives = alternatives;
+        }
+
+        public String getHelpMessage() {
+            String result = word + ": " + help + " [" + usage + "]";
+            return result;
+        }
+
+        public Collection<String> getKeys() {
+            Collection<String> words = new HashSet<>(alternatives);
+            words.add(word);
+            return words;
+        }
+    }
+
+    private static Logger logger = Logger.getLogger(LoginClientScannerProtocol.class);
+
+    public static boolean protocol(String message) {
+
+        List<String> words = splitMessage(message);
+
+        // If message is empty, then it tries to start, login or relogin
+        if(words.isEmpty()) {
```

Comment:
Eu percebo o teu problema, mas eu queria evitar estar sempre a escrever start e login :)
Se insistires eu apago... mas tens que insistir

Seja como for o loginclient serve apenas para testar